astroquery/simbad/core.py

```diff
@@ -355,316 +157,426 @@ def tap(self):
         """
-        Displays a description of the VOTable field.
+        tables = self.query_tap(query_tables)
+        tables["type"] = Column(["table"] * len(tables), dtype="object")
+        # the columns of basic are also valid options
+        basic_columns = self.list_columns("basic")[["column_name", "description"]]
+        basic_columns["column_name"].info.name = "name"
+        basic_columns["type"] = Column(["column of basic"] * len(basic_columns), dtype="object")
+        # get the bundles of columns from file
+        bundle_entries = {key: value for key, value in query_criteria_fields.items()
+                          if value["type"] == "bundle"}
+        bundles = Table({"name": list(bundle_entries.keys()),
+                         "description": [value["description"] for _, value in bundle_entries.items()],
+                         "type": ["bundle of basic columns"] * len(bundle_entries)},
+                        dtype=["object", "object", "object"])
+        # vstack the three types of options
+        return vstack([tables, basic_columns, bundles], metadata_conflicts="silent")
+
+    def _get_bundle_columns(self, bundle_name):
+        """Get the list of columns in the preselected bundles.
 
         Parameters
         ----------
-        field_name : str
-            the name of the field to describe. Must be one of those listed
-            by `list_votable_fields`.
+        bundle_name : str
+            The possible values can be listed with `~astroquery.simbad.SimbadClass.list_votable_fields`
 
-        Examples
-        --------
-        >>> from astroquery.simbad import Simbad
-        >>> Simbad.get_field_description('main_id')
-        main identifier of an astronomical object. It is the same as id(1)
-        >>> Simbad.get_field_description('bibcodelist(y1-y2)')
-        number of references. The parameter is optional and limit the count to
-        the references between the years y1 and y2
+        Returns
+        -------
+        list[Simbad.Column]
+            The list of columns corresponding to the selected bundle.
         """
-        # first load the dictionary from json
-        dict_file = get_pkg_data_filename(
-            os.path.join('data', 'votable_fields_dict.json'))
-        with open(dict_file, "r") as f:
-            fields_dict = json.load(f)
+        basic_columns = set(map(str.casefold, set(self.list_columns("basic")["column_name"])))
 
-        try:
-            print(fields_dict[field_name])
-        except KeyError:
-            raise KeyError("No such field_name")
+        bundle_entries = {key: value for key, value in query_criteria_fields.items()
+                          if value["type"] == "bundle"}
 
-    def get_votable_fields(self):
-        """
-        Display votable fields
+        if bundle_name in bundle_entries:
+            bundle = bundle_entries[bundle_name]
+            columns = [Simbad.Column("basic", column) for column in basic_columns
```

Comment:
Done, thanks.
They were in `SimbadClass` initially because these dataclasses tested that the columns and joins were valid. But I did not like the latency so I removed the tests and now the classes are self_sufficient :slightly_smiling_face: